gen/emit/EmitJulia.cpp

```diff
@@ -548,27 +592,194 @@ struct EmitJulia : public Emitter
+            for (auto& submod : moduleJlSubmodules) {
+                auto parts = StringUtil::split(submod, '.');
+                if (parts.size() >= 2) {
+                    // Restore each of the submodules we parsed from the
+                    // existing file (if any), and also generate the submodule
+                    // corresponding to the current package (if necessary)
+                    vector<string> parentParts(parts.begin(), parts.end() - 1);
+                    auto parent = StringUtil::join(parentParts, '.');
+                    auto module = parts.at(parts.size() - 1);
+                    fprintf(moduleJlFp, "@eval %s module %s ; end\n", parent.c_str(), module.c_str());
+                }
+            }
+            // LOAD_PATH controls where Julia looks for files you `import`.
+            // We're going to tell Julia that if we're in a package foo.bar.baz,
+            // it should first look in the `foo/` folder for any types it imports.
+            // unshift!(x, y) in Julia prepends y to the vector x. Its opposite
+            // is shift!(x) which removes the first element of x. We'll put that
+            // shift!(LOAD_PATH) in a `finally` block to ensure that the
+            // LOAD_PATH is restored even if something goes wrong with the
+            // imports
+            fprintf(moduleJlFp, "\nunshift!(LOAD_PATH, joinpath(@__DIR__, \"%s\"))\n", pdname.c_str());
+            fprintf(moduleJlFp, "try\n");
+            for (auto& import : moduleJlImports) {
+                // Restore each of the existing imports. Each import defines a
+                // single ZCM type somewhere in the package or one of its
+                // submodule
+                fprintf(moduleJlFp, "import %s", import.c_str());
+            }
+        }
+
+        ////////////////////////////////////////////////////////////
+        // STRUCTS
+        for (auto *ls_ : packStructs) {
+            auto& ls = *ls_;
+            string path = packageDir + "_" + ls.structname.nameUnderscore() + ".jl";
+
+            // If we're in a package, then add an appropriate import statement
+            // to ensure that this struct is added to the Julia module
+            if(moduleJlFp) {
+                fprintf(moduleJlFp, "import _%s\n", ls.structname.nameUnderscoreCStr());
+            }
+
+            if (!zcm.needsGeneration(ls.zcmfile, path))
+                continue;
+
+            EmitJulia E{zcm, ls, path};
+            if (!E.good()) return -1;
+            E.emitType();
+        }
+
+        if(moduleJlFp) {
+            // Restore LOAD_PATH no matter what
+            fprintf(moduleJlFp, "finally\n");
+            fprintf(moduleJlFp, "    shift!(LOAD_PATH)\n");
+            fprintf(moduleJlFp, "end\n");
+            fclose(moduleJlFp);
+            moduleJlFp = nullptr;
+        }
+
+        return 0;
+    }
+};
+
 int emitJulia(ZCMGen& zcm)
 {
     if (zcm.gopt->getBool("little-endian-encoding")) {
```

Comment:
What will it take to get this supported (probably more applicable for the main core-julia_support branch